### PR TITLE
Update GitWorkflow tutorial and improve the contributor's rst

### LINKED ISSUE #1877: "Git Workflow" tutorial outdated
http://seqan.readthedocs.io/en/master/Infrastructure/Contribute/GitWorkflow.html#infra-contribute-git
doesn't correctly describe our current git workflow, also some things should maybe be stated in CONTRIBUTING.rst more prominently.

Change the tutorial:
- [x] all bug-fixes, feature and other PRs go to the develop branch
- [x] only severe issues, such as packaging mistakes, regression and interface breakages shall results in PRs to master and an immediate patch-level release

### PR DESCRIPTION
**Pull request by h-2**
resolves #1877

### CODE REVIEW

File: manual/source/Infrastructure/Contribute/GitWorkflow.rst [hunk 1]
```diff
@@ -34,59 +34,63 @@ Execute the following command to get the last sources:
...
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Develop a feature or fix a bug
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Follow the `steps <https://www.atlassian.com/git/workflows#workflow-gitflow>`_ in “Mary and John begin new features” and “Mary finishes her feature”.
 
-* Create a new feature `branch <https://www.atlassian.com/git/tutorial/git-branches#branch>`_ based on `develop <https://github.com/seqan/seqan/tree/develop>`_.
+* Create a new `branch <https://www.atlassian.com/git/tutorial/git-branches#branch>`_ based on `develop <https://github.com/seqan/seqan/tree/develop>`_.
 * Perform your changes and `commit <https://www.atlassian.com/git/tutorial/git-basics#commit>`_ them onto your feature branch.
+* Keep your commit history lean (see below) and `write proper commit messages <infra-contribute-git-commits>`_.
```
Comment (rrahn) [inline manual/source/Infrastructure/Contribute/GitWorkflow.rst]: you mean *clean*

File: manual/source/Infrastructure/Contribute/GitWorkflow.rst [hunk 2]
```diff
@@ -34,59 +34,63 @@ Execute the following command to get the last sources:
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-* Create a new hotfix `branch <https://www.atlassian.com/git/tutorial/git-branches#branch>`_ based on `master <https://github.com/seqan/seqan/tree/master>`_.
-* Perform your changes and `commit <https://www.atlassian.com/git/tutorial/git-basics#commit>`_ them onto your hotfix branch.
-* When the fix is ready, push your hotfix branch to repository on GitHub. Then:
-    #. `Create a GitHub pull request`__ to `master <https://github.com/seqan/seqan/tree/master>`_.
-    #. `Create a GitHub pull request`__ to `develop <https://github.com/seqan/seqan/tree/develop>`_.
-    #. The pull requests should contain only the commits from your hotfix branch.
-* Delete your hotfix branch once it has been merged through the pull request.
+The git history of your contribution should be concise and lean. Please follow the following hints:
```
Comment (rrahn) [inline manual/source/Infrastructure/Contribute/GitWorkflow.rst]: *clean*???

Comment (h-2) [inline manual/source/Infrastructure/Contribute/GitWorkflow.rst]: http://pda.leo.org/ende/index_en.html#/info=167511 :D

**Comment by h-2** [comment]
http://pda.leo.org/ende/index_en.html#/info=167511 :D

File: manual/source/Infrastructure/Contribute/GitWorkflow.rst (hunk 2 above)
Comment (h-2) [inline manual/source/Infrastructure/Contribute/GitWorkflow.rst]: Wow the github mobile site sucks, sorry for the noise.

Comment (rrahn) [inline manual/source/Infrastructure/Contribute/GitWorkflow.rst]: 😒 I should go to bed earlier.

Comment (rrahn) [inline manual/source/Infrastructure/Contribute/GitWorkflow.rst]: Still concise implies lean then already, so it is kind of superfluous, or is this some phrase?

File: manual/source/Infrastructure/Contribute/GitWorkflow.rst (hunk 1 above)
Comment (rrahn) [inline manual/source/Infrastructure/Contribute/GitWorkflow.rst]: also here rather use concise than lean.

**Comment by h-2** [comment]
I like lean, but I have changed to concise :)

**Pull request merged**